### PR TITLE
Example script for ineterminate progress bar with a long running computation

### PR DESCRIPTION
**Pull request by GenevieveBuckley**
I found @tlambert03's [example script here](https://github.com/pyapp-kit/magicgui/issues/577#issuecomment-1705555387) very helpful for figuring out what I was doing wrong creating an indeterminate progress bar in magicgui

I thought it would be nice to add it to the examples in the repository, so ideally Talley does not have to repeat the same explanation again.

Closes https://github.com/pyapp-kit/magicgui/issues/577

### Summary by CodeRabbit

- New Feature: Added a new function `long_running` that performs a long-running computation with an indeterminate progress bar. This feature enhances user experience by providing visual feedback during lengthy computations.
- Documentation: Updated the heading in the README file under "Progress bar examples" for improved clarity.
- Test: Introduced two new test functions to verify the behavior of tqdm with an indeterminate total range both outside and inside the `magicgui` context, ensuring robustness of the new feature.

### CODE REVIEW

**Comment by GenevieveBuckley** [comment]
This might create a conflict with the other open PR https://github.com/pyapp-kit/magicgui/pull/571 but it should be fairly simple to resolve (I think all it should need when the time comes is:`git mv examples/progress_indeterminate.py docs/examples/progress_indeterminate.py`)

**Review by tlambert03 — APPROVED** [review]
Thanks!  Very helpful.

I'll get back to #571 soon as well and we can move this there then

File: docs/examples/progress_bars/progress_indeterminate.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        # It is best practice to use a separate thread for long running computations
+        # This makes the function non-blocking, you can still interact with the widget
+        @thread_worker(connect={"finished": lambda: pbar.progressbar.hide()})
+        def sleep(secs):
+            time.sleep(secs)
+
+        sleep(sleep_time)
+
+
+long_running.show(run=True)
```
[the remark this answers is not in the document]
Comment (GenevieveBuckley) [reply docs/examples/progress_bars/progress_indeterminate.py:31]: I don't agree that this suggestion is helpful. We can assume that users are capable of writing their own code, including error handling where needed. It also makes the example slightly less clear (we are trying to demonstrate the minimal requirements for an indeterminate progress bar, not this plus a bunch of other stuff).

[the remark this answers is not in the document]
Comment (tlambert03) [reply docs/examples/progress_bars/progress_indeterminate.py:31]: Totally agree.  You can ignore all these things.  I've found them interesting and somewhat helpful from time to time.  But I'll probably turn them off before too long.  Just experimenting for now :)

**Comment by tlambert03** [comment]
Thanks again!

**Pull request merged**